src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Utilities/IMsBuildAccessor.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Utilities
+{
+    /// <summary>
+    /// Utility class for allowing for testing of code that needs to access the msbuild lock, and also be testable.
+    /// </summary>
+    internal interface IMsBuildAccessor
+    {
+        /// <summary>
+        /// Gets the XML for a given unconfigured project.
+        /// </summary>
+        Task<string> GetProjectXml(UnconfiguredProject unconfiguredProject);
+
+        /// <summary>
+        /// Runs a given task inside either a read lock or a write lock.
+        /// </summary>
+        Task RunLocked(bool writeLock, Func<Task> task);
+    }
+
+    [Export(typeof(IMsBuildAccessor))]
+    [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
+    internal class MsBuildAccessor : IMsBuildAccessor
+    {
+        private readonly IProjectLockService _projectLockService;
+
+        [ImportingConstructor]
+        public MsBuildAccessor(IProjectLockService projectLockService)
+        {
+            _projectLockService = projectLockService;
+        }
+
+        public async Task<string> GetProjectXml(UnconfiguredProject unconfiguredProject)
+        {
+            var configuredProject = await unconfiguredProject.GetSuggestedConfiguredProjectAsync().ConfigureAwait(false);
```

Comment:
@lifengl? I just assumed that would give us the active based on this:https://github.com/Microsoft/VSProjectSystem/blob/master/doc/automation/finding_CPS_in_a_VS_project.md